src/qtils/move.hpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <utility>
+
+namespace qtils {
+  /**
+   * Used to reset `std::coroutine_handle` on move.
+   */
+  template <typename T>
+  struct MoveOnly {
+    MoveOnly() : v{} {};
```

Comment:
```suggestion
    MoveOnly() = default;
```
'cause T might not have default ctor

Reply:
No, `T` must have default constructor